### PR TITLE
Revamp login and registration configurations

### PR DESCRIPTION
**Pull request by NipuniBhagya**
### Related Issues
- https://github.com/wso2/product-is/issues/17156
- https://github.com/wso2/product-is/issues/17590
- https://github.com/wso2/product-is/issues/17901

### Preview

image

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/extensions/components/password-history-count/components/password-history-count.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export const generatePasswordHistoryCount = (
                 <Icon name="info circle" />
                 { t("extensions:manage.serverConfigurations.passwordHistoryCount.message") }
             </Message>
-            <Divider hidden />
+            <Divider className="mt-6 mb-6" />
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/components/password-history-count/components/password-history-count.tsx:75]: AFAIK, we are getting away from semantic ui classes. Let's use Oxygen UI divider and styles here if possible

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    ShieldUserPencilIcon, 
+    UserDatabaseIcon, 
+    UserDocumentIcon, 
+    UserGearIcon, 
+    UserPlusIcon 
+} from "@oxygen-ui/react-icons";
+import { ServerConfigurationsConstants } from "../constants/server-configurations-constants";
+import Card from "@oxygen-ui/react/Card";
+import { CardContent } from "@oxygen-ui/react";
+import { serverConfigurationConfig } from "apps/console/src/extensions";
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:44]: Let's use relative path imports and avoid importing from higher levels

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        dynamicConnectors 
+    } = props;
+
+    const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+    /**
+     * Handles template selection.
+     *
+     * @param e - Click event.
+     * @param id - Id of the template.
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:84]: Doc comment contradicts with the function signature

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+                            color="text.primary"
+                            className="mb-3" 
+                            variant="h4"
+                        >
+                            { category?.title }
+                        </Typography>
+                        <div className="governance-connector-list-grid-wrapper" data-componentid={ componentId }>
+                            <div className="governance-connector-list-grid">
+                                {
+                                    category.connectors.map((connector: any) => {
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:194]: Let's fix the other places as well.

```suggestion
                                    category?.connectors?.map((connector: any) => {
```

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+                                                    </CardContent>
+                                                    <CardContent>
+                                                        <Typography variant="body2" color="text.secondary">
+                                                            { connector.description }
+                                                        </Typography>
+                                                    </CardContent>
+                                                </Card>
+                                            );
+                                        }
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:220]:
```suggestion
```

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+                                        
+                                    })
+                                }
+                            </div>
+                        </div>
+                    </div>                  
+                );
+            }) }
+            {
+                (serverConfigurationConfig.dynamicConnectors && dynamicConnectors.length > 0) && (
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:229]:
```suggestion
                (serverConfigurationConfig.dynamicConnectors?.dynamicConnectors?.length > 0) && (
```

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+                    <div className="catergory-container">
+                        <Typography 
+                            color="text.primary"
+                            className="mb-3" 
+                            variant="h4"
+                        >
+                            Other Settings
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:236]: Let's i18n these strings

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+};
+
+/**
+ * Default props for the component.
+ */
+GovernanceConnectorCategoriesGrid.defaultProps = {
+    "data-componentid": "governance-connector-category-grid"
+};
+
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:284]:
```suggestion
```

File: apps/console/src/features/server-configurations/components/governance-connector-grid.tsx [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+            .replace(":categoryId", 
+                connector.categoryId)
+            .replace(":connectorId", connector.id));
+    };
+
+    if (isLoading) {
+        return <ContentLoader dimmer />;
+    }
+
+    const resolveConnectorCategoryIcon = (id : string): ReactElement => {
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/components/governance-connector-grid.tsx:102]: Will it be possible to reuse connectorIcon object defined in apps/console/src/features/server-configurations/configs/ui.ts

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface } from "@wso2is/core/models";
+import { Field, Form } from "@wso2is/form";
+import isEmpty from "lodash-es/isEmpty";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:24]:
```suggestion
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 11]
```diff
@@ -0,0 +1,260 @@
...
+ * Interface for admin forced password reset form props.
+ */
+interface AdminForcedPasswordResetFormPropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Connector's initial values.
+     */
+    initialValues: GovernanceConnectorInterface;
+    /**
+     * Callback for form submit.
+     * @param values - Resolved Form Values.
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:40]:
```suggestion

     * @param values - Resolved Form Values.
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+        isSubmitting,
+        ["data-componentid"]: testId
+    } = props;
+
+    const { t } = useTranslation();
+
+    const [ initialConnectorValues, setInitialConnectorValues ]
+        = useState<Map<string, ConnectorPropertyInterface>>(undefined);
+    const [ initialFormValues, setInitialFormValues ]
+        = useState<any>(undefined);
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:83]:
```suggestion
    const [ initialFormValues, setInitialFormValues ]
        = useState<Map<string,FormValue>>(undefined);
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+
+    /**
+     * Flattens and resolved form initial values and field metadata.
+     */
+    useEffect(() => {
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:89]:
```suggestion
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 14]
```diff
@@ -0,0 +1,260 @@
...
+    };   
+   
+    if (!initialConnectorValues) {
+        return null;
+    }
+
+    return (
+        <Form
+            id={ FORM_ID }
+            uncontrolledForm={ true }
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:140]:
```suggestion
            uncontrolledForm
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx [hunk 15]
```diff
@@ -0,0 +1,260 @@
...
+                defaultValue={ initialFormValues?.[ 
+                    "Recovery.AdminPasswordReset.RecoveryLink" ] === "true" }
+                readOnly={ readOnly }
+                disabled={ !isConnectorEnabled }
+                width={ 16 }
+                data-componentid={ `${ testId }-enable-auto-login` }
+                hint={ GovernanceConnectorUtils.resolveFieldHint(
+                    "Account Management",
+                    "Recovery.AdminPasswordReset.RecoveryLink", 
+                    "User gets notified with a link to reset password.")
```
Comment (pavinduLakshan) [inline apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:164]: Shouldn't we need to i18n these strings?

File: apps/console/src/features/server-configurations/configs/ui.ts [hunk 16]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
+ * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com).
```
Comment (jeradrutnam) [inline apps/console/src/features/server-configurations/configs/ui.ts:2]:
```suggestion
 * Copyright (c) 2020-2023, WSO2 LLC. (https://www.wso2.com).
```

File: apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx (hunk 15 above)
Comment (NipuniBhagya) [reply apps/console/src/features/server-configurations/forms/admin-forced-password-reset.tsx:164]: Will provide the i18n placeholder strings in a followup PR

**Review by DonOmalVindula — DISMISSED** [review]

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**